lib/spack/spack/installer.py

```diff
@@ -412,6 +413,20 @@ def clear_failures():
     spack.store.db.clear_all_failures()
 
 
+def combine_phase_logs(pkg):
+    """
+    Each phase will produce it's own log, so this function aims to cat all the
+    separate phase log output files into the pkg.log_path.
+    Args:
+        pkg (Package): the package that was built and installed
+    """
+    log_files = pkg.phase_log_files
+
+    with open(pkg.log_path, 'w') as fd:
+        for line in itertools.chain.from_iterable(list(map(open, log_files))):
```

Comment:
are these ever closed?

Reply:
Totally missed this. No I don't think so.